x-pack/plugin/security/qa/operator-privileges-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/operator/Constants.java

```diff
@@ -190,9 +192,9 @@ public class Constants {
         "cluster:admin/xpack/ml/upgrade_mode",
         "cluster:admin/xpack/monitoring/bulk",
         "cluster:admin/xpack/monitoring/migrate/alerts",
-        "cluster:admin/xpack/query_rules/delete",
-        "cluster:admin/xpack/query_rules/get",
-        "cluster:admin/xpack/query_rules/put",
+        new FeatureFlag("query_rules").isEnabled() ? "cluster:admin/xpack/query_rules/delete" : null,
+        new FeatureFlag("query_rules").isEnabled() ? "cluster:admin/xpack/query_rules/get" : null,
+        new FeatureFlag("query_rules").isEnabled() ? "cluster:admin/xpack/query_rules/put" : null,
```

Comment:
The [`query_rules`](https://github.com/elastic/elasticsearch/blob/98e3230e4be123a98e46051b10477203f51bb885/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java#L112) feature flag is private in the enterprise search plugin. I don't want to touch production code for this fix. So I just instantiate the feature flag again since underlyingly it just reads a system property.